src/commands/synthetics/run-tests-command.ts

```diff
@@ -366,7 +369,9 @@ export class RunTestsCommand extends Command {
         startUrl: validatedOverrides.startUrl,
         startUrlSubstitutionRegex: validatedOverrides.startUrlSubstitutionRegex,
         testTimeout: validatedOverrides.testTimeout,
-        variables: parseVariablesFromCli(this.variableStrings, (log) => this.reporter.log(log)),
+        // TODO SYNTH-12989: Clean up deprecated `variableStrings` in favor of `variables` in `defaultTestOverrides`.
+        variables:
+          validatedOverrides.variables ?? parseVariablesFromCli(this.variableStrings, (log) => this.reporter.log(log)),
```

Comment:
Can you also please add a test for checking the priority behaviour here.

Reply:
Hey and as we discussed also one for checking the old behaviour still works as well

Reply:
and also for the override priority :D 
here - https://github.com/DataDog/datadog-ci/pull/1354/files#diff-fed723dc96fbca46fa0fc4add02e5a092a935c4778780668acc07e993bc916deR375
and here - https://github.com/DataDog/datadog-ci/pull/1354/files#diff-fed723dc96fbca46fa0fc4add02e5a092a935c4778780668acc07e993bc916deR415

i'm okay if you end up creating a card for this one as it will require adding cases not just for variables but also all the other parameters we've added to env and cli 👍